File: rb/features.rb
# GeoadminSearch SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module GeoadminSearchFeatures
  def self.make_feature(name)
    case name
    when "base"
      GeoadminSearchBaseFeature.new
    when "ratelimit"
      GeoadminSearchRatelimitFeature.new
    when "retry"
      GeoadminSearchRetryFeature.new
    when "test"
      GeoadminSearchTestFeature.new
    when "timeout"
      GeoadminSearchTimeoutFeature.new
    else
      GeoadminSearchBaseFeature.new
    end
  end
end

File: py/features.py
# GeoadminSearch SDK feature factory

from feature.base_feature import GeoadminSearchBaseFeature
from feature.test_feature import GeoadminSearchTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GeoadminSearchBaseFeature(),
        "test": lambda: GeoadminSearchTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
